gstreamer-pbutils/src/auto/encoding_profile.rs

```diff
@@ -0,0 +1,265 @@
+// This file was generated by gir (d933f9a+) from gir-files (???)
+// DO NOT EDIT
+
+use DiscovererInfo;
+use ffi;
+use glib;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct EncodingProfile(Object<ffi::GstEncodingProfile, ffi::GstEncodingProfileClass>);
+
+    match fn {
+        get_type => || ffi::gst_encoding_profile_get_type(),
+    }
+}
+
+impl EncodingProfile {
+    pub fn find<'a, P: Into<Option<&'a str>>>(targetname: &str, profilename: &str, category: P) -> Option<EncodingProfile> {
+        assert_initialized_main_thread!();
+        let category = category.into();
+        let category = category.to_glib_none();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_profile_find(targetname.to_glib_none().0, profilename.to_glib_none().0, category.0))
+        }
+    }
+
+    pub fn from_discoverer(info: &DiscovererInfo) -> Option<EncodingProfile> {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_profile_from_discoverer(info.to_glib_none().0))
+        }
+    }
+}
+
+unsafe impl Send for EncodingProfile {}
+unsafe impl Sync for EncodingProfile {}
+
+pub trait EncodingProfileExt {
+    fn copy(&self) -> Option<EncodingProfile>;
+
+    fn get_allow_dynamic_output(&self) -> bool;
+
+    fn get_description(&self) -> Option<String>;
+
+    fn get_file_extension(&self) -> Option<String>;
+
+    fn get_format(&self) -> Option<gst::Caps>;
+
+    fn get_input_caps(&self) -> Option<gst::Caps>;
+
+    fn get_name(&self) -> Option<String>;
+
+    fn get_presence(&self) -> u32;
+
+    fn get_preset(&self) -> Option<String>;
+
+    fn get_preset_name(&self) -> Option<String>;
+
+    fn get_restriction(&self) -> Option<gst::Caps>;
+
+    fn get_type_nick(&self) -> Option<String>;
+
+    fn is_enabled(&self) -> bool;
+
+    fn set_allow_dynamic_output(&self, allow_dynamic_output: bool);
+
+    fn set_description(&self, description: &str);
```

Comment:
If description above can be NULL, this should be an Option. Same thing for other functions